### PR TITLE
sail: sail serves the correct assets for the requested room [ch2283]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @nicks, @jazzdan,

Please review the following commits I made in branch maiamcc/sail-serve-right-version:

5354bb72b31ff3be23609d1db1c50c2f007b71a3 (2019-05-02 18:36:42 -0400)
clean-up

b721f358e26d563261e8374a5a05a7c288198b47 (2019-05-02 18:28:40 -0400)
inject version into returned html

744c3f5f0a6ecb5c452040eea3415f895f12f3db (2019-05-02 18:28:35 -0400)
respect versions in prod server urls

0a1daa52a902c1198415e662b5800174253ec593 (2019-05-02 18:28:32 -0400)
wip

0bf54aa51c65cb443511f995bfcb475485e266a6 (2019-05-02 18:28:30 -0400)
send version with new_room request

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: deployments/sail.dockerfile [hunk 1]
```diff
@@ -19,4 +19,4 @@ RUN make build-js
 ADD . .
 RUN make install-sail
 
-ENTRYPOINT sail --web-mode=precompiled
+ENTRYPOINT sail --web-mode=prod
```
Comment (nicks) [inline deployments/sail.dockerfile]: if we're doing web-mode prod, we probably don't need to include the JS anymore? can get rid of the `make build-js` line above

File: internal/assets/server.go [hunk 2]
```diff
@@ -26,7 +28,9 @@ import (
 )
 
 const prodAssetBucket = "https://storage.googleapis.com/tilt-static-assets/"
-const webVersionKey = "web_version"
+const WebVersionKey = "web_version"
+
+var versionRe = regexp.MustCompile(`/(v\d*\.\d*\.\d*)/.*`) // matches `/vA.B.C/...`
```
Comment (nicks) [inline internal/assets/server.go]: should probably be \d+ in place of \d*

File: internal/sail/server/server.go [hunk 3]
```diff
@@ -51,7 +53,25 @@ func (s SailServer) newRoom(w http.ResponseWriter, req *http.Request) {
...
+	}
+
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error reading request: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	var newRoomReq model.SailNewRoomRequest
+	err = json.Unmarshal(body, &newRoomReq)
```
Comment (nicks) [inline internal/sail/server/server.go]: it'd be more idiomatic to do json.NewDecoder(req.Body).Decode(&newRoomReq)

and save yourself some error handling

File: internal/assets/server.go (hunk 2 above)
Comment (maiamcc) [reply internal/assets/server.go]: good catch!

File: internal/sail/server/server.go (hunk 3 above)
Comment (maiamcc) [reply internal/sail/server/server.go]: i never knew! thanks :)

File: deployments/sail.dockerfile (hunk 1 above)
Comment (maiamcc) [reply deployments/sail.dockerfile]: oh shoot this was just for testing and I left it in

**Pull request merged**